MekHQ/src/mekhq/campaign/mission/resupplyAndCaches/Resupply.java

```diff
@@ -862,42 +880,45 @@ private void calculatePlayerConvoyValues() {
         totalPlayerCargoCapacity = 0;
 
         for (Force force : campaign.getAllForces()) {
-            if (!force.isConvoyForce()) {
+            if (!force.getForceType().isConvoy()) {
```

Comment:
I think I like more a `force.getForceType().isConvoy()` as you already have the force here and the api already exists, so you only change the internals of the force object, not its API.

Reply:
I'm not entirely sure what you mean by that, sorry

Reply:
```suggestion
            if (!force.isConvoy()) {
```

you already had the function `boolean isConvoy()` in the class api, the change you made in the force type didnt required you to change the API of the object, which is a benefit of the object oriented design.

Reply:
Ahhhh, yes, I see exactly what you mean! Sorry, the use of 'API' here threw me off as that's not a term I'm super familiar with in this context.

Reply:
API just mean the "application interface" or something like that, the api of a class are all its public methods. The method is already there, so there is no need to change how the "world" interacts with it, if the world needs to know if a force is a convoy or not, let the world ask the force if its a convoy, internally the force deals with the question however it wants, without exposing its internal workings to the outside.

Reply:
That makes a heap of sense, thank you :)